### PR TITLE
feat(metrics) Low cardinality limit for projects setup

### PR DESCRIPTION
**Pull request by vgrozdanic**
Limit projects cardinality limit for testing purposes.

Instead of hitting the real cardinality limits on relay, we can configure low cardinality limits for them, and test what is happening when cardinality limits are set and how to present it to the user.

### CODE REVIEW

**Comment by ArthurKnaus** [comment]
@vgrozdanic If we want to see cardinality warnings in the frontend, we need to adjust the constant:
https://github.com/getsentry/sentry/blob/aa98fdae48c0709f26e16657152084f671c7fbd3/static/app/utils/metrics/constants.tsx#L16

**Review by iambriccardo — APPROVED** [review]
LGTM, are we sure these numbers are what we want for testing? It might make the testing easier if we could configure the limit via an option, but only if we are not sure about `100`.

I will let @Dav1dde take a final look since he knows best cardinality limiting

**Review by jjbayer — COMMENTED** [review]
Instead of overwriting limits defined in `sentry-metrics.cardinality-limiter.limits.{namespace.value}.per-org` with hardcoded values, should we instead define a second option `sentry-metrics.cardinality-limiter.limits.custom.per-org.low`, and then select the right option based on the feature flag?

**Comment by vgrozdanic** [comment]
@ArthurKnaus we also want the relay to cardinality limits us so that we can query outcomes metric: [#74258](https://github.com/getsentry/sentry/issues/74258)

**Comment by ArthurKnaus** [comment]
> @ArthurKnaus we also want the relay to cardinality limits us so that we can query outcomes metric: [#74258](https://github.com/getsentry/sentry/issues/74258)

Perfect, just wanted to make sure 👍

**Comment by vgrozdanic** [comment]
@iambriccardo  @jjbayer initial plan was that this wouldn't be used for a long time, only for a short-lived development of the UI around cardinality limits. But when I gave it a second thought, it makes more sense to add extra option and based on the feature flag pick one, since we might need it in the future development :)

PR with options will be merged before this one: https://github.com/getsentry/sentry-options-automator/pull/1947/files

File: src/sentry/relay/config/__init__.py [hunk 1]
```diff
@@ -241,7 +241,16 @@ def get_metrics_config(timeout: TimeChecker, project: Project) -> Mapping[str, A
     cardinality_limits: list[CardinalityLimit] = []
     for namespace in CARDINALITY_LIMIT_USE_CASES:
         timeout.check()
-        option = options.get(f"sentry-metrics.cardinality-limiter.limits.{namespace.value}.per-org")
+        if (
+            features.has("organizations:low-cardinality-limits-on-project", project.organization)
+            and namespace == UseCaseID.CUSTOM
+        ):
+            option = options.get("sentry-metrics.cardinality-limiter.limits.custom.per-org.low")
```
Comment (jjbayer) [inline src/sentry/relay/config/__init__.py:248]: You'll need to register this option in `sentry.options.default.

nit: Does it make sense to save a per-project limit in an option called "per-org"? I like the idea of putting the scope in the option, but maybe we should remove `per-org` from the option name.

**Comment by jjbayer** [comment]
> PR with options will be merged before this one

@vgrozdanic you'll need to register the option before setting it, but you can set the desired values as default when registering the option to make sure that you have sensible settings once you merge this PR (strictly speaking you don't even need the automator PR, but it's good to set the values explicitly).

File: src/sentry/relay/config/__init__.py (hunk 1 above)
Comment (vgrozdanic) [reply src/sentry/relay/config/__init__.py]: I agree, options is renamed and I have added it to defaults file

**Comment by vgrozdanic** [comment]
@jjbayer I can close the other PR to add options to automator, but maybe it's better if we also have it as explicit option so that can more easily be configured without the need to deploy everything if we just want to change the limit to something smaller/larger

**Review by jjbayer — APPROVED** [review]

File: src/sentry/relay/config/__init__.py [hunk 2]
```diff
@@ -256,7 +265,7 @@ def get_metrics_config(timeout: TimeChecker, project: Project) -> Mapping[str, A
                 "granularitySeconds": quota["granularity_seconds"],
             },
             "limit": quota["limit"],
-            "scope": "organization",
+            "scope": options.get("scope", "organization"),
```
Comment (Dav1dde) [inline src/sentry/relay/config/__init__.py:268]: You query a global `scope` option for this?

**Review by Dav1dde — CHANGES_REQUESTED** [review]
Let's not do this.

- The option your modifying is only there for legacy reasons (compatibility with the old cardinality limiter). You're adding more cruft to it.
- We don't need a feature flag for this

I showed you this yesterday, add your project allow list as another field/filter to here: https://github.com/getsentry/sentry/blob/5a93c4016808d1e935b073becc0856b05b200bc8/src/sentry/relay/config/__init__.py#L228-L231

Then you can freely configure limits as you please for testing.

File: src/sentry/relay/config/__init__.py (hunk 2 above)
Comment (vgrozdanic) [reply src/sentry/relay/config/__init__.py]: Nice catch, thank you

Comment (jjbayer) [reply src/sentry/relay/config/__init__.py]: Whoops

**Comment by jjbayer** [comment]
> The option your modifying is only there for legacy reasons (compatibility with the old cardinality limiter). You're adding more cruft to it.

@Dav1dde when reviewing this I wasn't aware that these are legacy options. Could you please clarify that in code somehow? E.g. split out into a function `_get_legacy_limits()` or just add a bunch of comments.

**Comment by Dav1dde** [comment]
@jjbayer as with a lot of things this is on my todo to cleanup :(

**Comment by vgrozdanic** [comment]
@jjbayer @Dav1dde changes applied

**Review by Dav1dde — APPROVED** [review]

**Pull request merged**